lib/snmp.js

```diff
@@ -676,21 +677,37 @@ Session.prototype.getAll = function (options, callback) {
         }
 
         self.sendMsg(pkt, options, function (err, varbinds) {
-            if (options.abortOnError && err) {
-                callback(err);
-            } else {
-                if (varbinds) {
-                    results = results.concat(varbinds);
-                }
-                if (c < options.oids.length) {
-                    getOne(c);
+            if (!deadlineExpired) {
```

Comment:
Perhaps an `if (deadlineExpired) { return; }` here instead, to avoid the indentation? I don't think the extra check below is required either given that we do this check at the top level?